dev-docs/rfcs/001_sudo_mode/sudo_mode.md

```diff
@@ -0,0 +1,104 @@
+- The session duration is determined by `User#session_duration_seconds` which defaults to 30 days https://github.com/hackclub/hcb/blob/d293e57763729f69a1612606e22095761650eccd/db/schema.rb#L2173
+  - This value is user-configurable from a pre-defined set of options https://github.com/hackclub/hcb/blob/ad940d9b29be634ff412b6f88f1a8e58bb624444/app/views/users/edit.html.erb#L222-L225
+  - We don't currently validate this value (see https://github.com/hackclub/hcb/pull/10660)
+  - When a session is created we compute the expiry time using this value and set it on both the cookie and `UserSession` https://github.com/hackclub/hcb/blob/8508ac95625ca08aef11b7919596ddb8f68c0665/app/helpers/sessions_helper.rb#L28
+- The session duration is fixed upon creation and isn't affected by user activity. There is interest in changing this: https://github.com/hackclub/hcb/issues/7258
+  - We do however update `UserSession#last_seen_at` on every request https://github.com/hackclub/hcb/blob/8508ac95625ca08aef11b7919596ddb8f68c0665/app/controllers/application_controller.rb#L21-L22
+- We don't currently clear out older sessions but this may be something worth looking into ([internal discussion](https://hackclub.slack.com/archives/C047Y01MHJQ/p1750259883680629)).
+- `UserSession` creation is tracked using the [`public_activity` gem](https://github.com/public-activity/public_activity) so it can be displayed under "Recent activity"
+
+    <img src="login_activity.png" width="200"/>
+    
+    This is configured here https://github.com/hackclub/hcb/blob/cbfccf8975fd8a9c258f916034ef8e7836e80229/app/models/user_session.rb#L47-L48 and displayed here https://github.com/hackclub/hcb/blob/d293e57763729f69a1612606e22095761650eccd/app/controllers/my_controller.rb#L6-L13
+
+## Design considerations
+
+> [!NOTE]
+> We will assume a user is in sudo mode if they are logged in and last authenticated less than 2 hours ago
+
+```mermaid
+flowchart LR
+    A["Log in"] -- Time passes --> B["Initiate sudo action"]
+    B --> C["Authentication older than 2 hours?"]
+    C -- Yes --> D["Re-authenticate"]
+    D --> E["Proceed"]
+    C -- No --> E
+```
+
+**Assumptions**
+
+1. Users who initiate a sudo action and subsequently abandon it (e.g. because they don't have access to their email or TOTP) should remain logged in.
+2. Re-authentication only requires a single factor of the user's choosing
+3. Once they are re-authenticated we won't require them to do so for another two hours
+
+**Open questions**
+
+1. We currently show users a list of their sessions including how long ago they logged in. Should we also surface information about re-authentications?
+
+   <img src="sessions.png" width="200"/>
+2. We send users an email when there is a new login on their account. Would we want to do the same for re-authentications?
```

Comment:
We normally only do this if the context is unrecognised, I think it's ok to not do this